lte/gateway/c/core/oai/lib/secu/key_nas_deriver.c

```diff
@@ -71,3 +71,212 @@ int derive_key_nas(
   memcpy(knas, &out[31 - 16 + 1], 16);
   return 0;
 }
+
+int derive_5gkey_gnb(const uint8_t* kamf, uint32_t ul_count, uint8_t* kgnb) {
+  uint8_t s[10]   = {0};
+  uint8_t out[32] = {0};
+
+  /*
+   * FC
+   */
+  s[0] = 0x6E;
+  /*
+   * P0 = serving network name
+   */
+  s[1] = (ul_count >> 24) & 0xFF;
+  s[2] = (ul_count >> 16) & 0xFF;
+  s[3] = (ul_count >> 8) & 0xFF;
+  s[4] = ul_count & 0xFF;
+  /*
+   * L0 = length(P0) = 4
+   */
+  s[5] = 0x00;
+  s[6] = 0x04;
+  /*
+   * P1 = Access type distinguisher 3GPP access
+   */
+  s[7] = 0x01;
+  /*
+   * L1 = length(P1) = 1
+   */
+  s[8] = 0x00;
+  s[9] = 0x01;
+
+  kdf(kamf, 32, &s[0], 10, &out[0], 32);
+  memcpy(kgnb, &out[0], 32);
+  return 0;
+}
+
+int derive_5gkey_ausf(
```

Comment:
no longer needed, this is pushed to subscriberdb side

Reply:
comment addressed.
removed derive_5gkey_ausf